b2share/modules/communities/models.py

```diff
@@ -134,7 +137,11 @@ def receive_before_insert(mapper, connection, target):
             community=str(target.id),
             old_state=PublicationStates.submitted.name,
             new_state=PublicationStates.draft.name
-        )
+        ),
+        # permission to update the metadata of a published record
+        update_record_metadata_need_factory(
+            community=str(target.id),
+        ),
```

Comment:
I just tested your PR with this command and gave the permission to my user:
`b2share access allow -e my@email.com 'update-record-metadata' -a '{"community":"e9b9792e-79fb-4b07-b6b4-b9c2bd06d095"}'`
It is possible to do something similar for roles.